src/lib/provable/scalar-field.ts

```diff
@@ -0,0 +1,44 @@
+import { createField } from './core/field-constructor.js';
+import { createForeignField } from './foreign-field.js';
+import { field3ToShiftedScalar } from './gadgets/native-curve.js';
+import { Provable } from './provable.js';
+import { Scalar } from './scalar.js';
+
+export { ScalarField };
+
+/**
+ * ForeignField representing the scalar field of Pallas and the base field of Vesta
+ */
+class ScalarField extends createForeignField(
```

Comment:
to make this class more usable, I recommend adding wrappers for the most important static methods, at least `.from()`, to make it return a `ScalarField`

Reply:
Unfortunately, that is not possible due the overloading of the base `.from()`. I added the static method with a `ForeignField` as parameters, I think that should give enough convenience in constructing and working with them.

Reply:
right, because the `.from()` return is canonical! ok